File: JavaScript/7js 3/workNeeded.js
/*
|--------------------------------------------------------------------------
| Offload your work!
|--------------------------------------------------------------------------
|
*/

function workNeeded(projectMinutes, freelancers) {
   let x = projectMinutes - freelancers.reduce((a, b) => a + b[0] * 60 + b[1], 0);
   return x <= 0 ? "Easy Money!" : `I need to work ${(x / 60) | 0} hour(s) and ${x % 60} minute(s)`;
}


const workNeeded = (p, f) =>
   (p = f.reduce((s, [h, m]) => s - h * 60 - m, p)) <= 0
      ? "Easy Money!"
      : `I need to work ${(p / 60) | 0} hour(s) and ${p % 60} minute(s)`;


function workNeeded(project_minutes, freelancers) {
   let i;
   for (i = 0; i < freelancers.length; i += 1)
      project_minutes -= freelancers[i][0] * 60 + freelancers[i][1];
   return project_minutes <= 0
      ? "Easy Money!"
      : "I need to work " +
           Math.floor(project_minutes / 60) +
           " hour(s) and " +
           String(project_minutes % 60) +
           " minute(s)";
}


console.log(workNeeded(60, [[1, 0]]), "Easy Money!");
console.log(workNeeded(60, [[0, 0]]), "I need to work 1 hour(s) and 0 minute(s)");
console.log(
   workNeeded(141, [
      [1, 55],
      [0, 25],
   ]),
   "I need to work 0 hour(s) and 1 minute(s)"
);

File: Python/7py 3/work_needed.py
"""
|--------------------------------------------------------------------------
| Offload your work!
|--------------------------------------------------------------------------
|
"""

def work_needed(project_minutes, free_lancers):
   x = project_minutes - sum(time[0] * 60 + time[1] for time in free_lancers)
   return "Easy Money!" if x <= 0 else f'I need to work {x // 60} hour(s) and {x % 60} minute(s)'


work_needed = lambda p,f:(lambda t:("Easy Money!","I need to work %d hour(s) and %d minute(s)"%divmod(t,60))[t>0])(p-sum(h*60+m for h,m in f))


def work_needed(mins, fls):
   time = mins - sum ([h*60 + m for h,m in fls])
   h, m = int(time / 60), time%60
   return 'I need to work {} hour(s) and {} minute(s)'.format(h,m) if time > 0 else 'Easy Money!'


print(work_needed(60, [(1,0)]), "Easy Money!")
print(work_needed(60, [(0,0)]), "I need to work 1 hour(s) and 0 minute(s)")
print(work_needed(141, [(1,55), (0,25)]), "I need to work 0 hour(s) and 1 minute(s)")